src/utils.js

```diff
@@ -45,8 +45,60 @@ function getValue(value) {
     return VALUE_EXTRACT_MAP[value.type](value);
 }
 
+/*
+ * Accumulates results from a Yamcs API request that does paging
+ * with a continuation token. The Yamcs responses generally contain
+ * an indication of the total size, a continuation token, if there
+ * are more results, and another property that is an array of results
+ * for the current request. That array is what is to be accumulated
+ * over multiple requests. The property name for that array is a
+ * parameter to this function.
+ *
+ * Parameters:
+ *     url          the API URL
+ *     property     the property of the JSON responses to be gathered
+ *     totalLimit   (optional) a limit on the total number of objects
+ *                  to gather over multiple requests
+ *     token        (optional) the continuation token value from
+ *                  a prior request
+ *
+ * Returns:
+ *     a promise for an array of results accumulated over the requests
+ */
+function accumulateResults(url, property, soFar, totalLimit, token) {
```

Comment:
@merose I think that's a good short-term solution for the telemetry tree. I'm not sure how sustainable it will be for large dictionaries, or as a general solution for fetching large amounts of telemetry. We regularly see telemetry dictionaries with 50000+ for some of our missions. If each query is limited to 1000 results, that's 50 queries just to build the dictionary. This might take 30 seconds or more.

To address this we might consider progressively loading the tree on demand. This would cause a problem for search which currently uses an in-memory index built by eagerly traversing the entire object tree, but we could potentially hand off search to the server if YAMCS supports it. Requires a followup issue and some investigation of the YAMCS API.

Reply:
That makes sense. Yamcs currently supports some query capabilities when getting parameter info, but it may or may not be sufficient for incrementally getting the object tree. We'll have to discuss further and try some experiments. If the existing capabilities are not sufficient, we request changes of Yamcs.